lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -220,12 +338,18 @@ export class ImageSelector extends React.Component<
 			return;
 		}
 
-		const source = new sdk.sourceDestination.File({
-			path: imagePath,
-		});
+		let source;
+		if (SourceType === sourceDestination.File) {
+			source = new sourceDestination.File({
+				path: imagePath,
+			});
+		} else {
+			source = new sourceDestination.Http(imagePath);
+		}
+
 		try {
 			const innerSource = await source.getInnerSource();
-			const metadata = (await innerSource.getMetadata()) as sdk.sourceDestination.Metadata & {
+			const metadata = (await innerSource.getMetadata()) as sourceDestination.Metadata & {
```

Comment:
We need the UI to show that we are loading the metadata, maybe don't close the URL selector until we have the metadata.

Reply:
Made it so that the button is disabled and it closes when the data's done loading